common/src/main/java/io/pravega/common/lang/BigLong.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.lang;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.util.BitConverter;
+import lombok.Data;
+
+@Data
+/**
+ * This class represents a 96 bit number with 32 bit msb encoded as integer and 64 bit lsb encoded as long.
+ * It is unsigned and only allows for non negative values. It also implements comparable interface and comparison involves
+ * first compariging msbs and if msbs are equal then we compare lsbs.
+ */
+public class BigLong implements Comparable {
```

Comment:
I think the name of this could be better. It sounds like it should be related to Java's BigInteger. 
If it were 128 bits I would say call it "LongLong". So maybe it should be "Int96" or something? Any ideas?

Reply:
I like `Int96`. `LongLong` sounds like 2x long, which is not what this class is encapsulating.

Reply:
+1 for `Int96`. It is a really descriptive name. `BigLong` might confuse people to think it's 128 bit.